### DISCUSSION TITLE
No sound with vlc or mpv

### DISCUSSION THREAD

**Discussion by realpelou** [streamlink/streamlink#5631] [Q&A] [answered]
### Description

Hello,

Sorry, I don't know if this is the right place or not to ask this question.

A friend of mine installed streamlink on my computer over a year ago so that I could watch a channel and record it with VLC. I hadn't been using it for a few months, but yesterday when I tried it, it stopped working. Here's the command I used and the error.

```
streamlink --ffmpeg-ffmpeg "C:\ffmpeg\bin\ffmpeg.exe"  https://www.tf1.fr/lci/direct 720p
[cli][info] Found matching plugin tf1 for URL https://www.tf1.fr/lci/direct
[cli][info] Available streams: 720p_hls, 576p_hls, 360p_hls, 234p_hls, 234p (worst), 360p, 576p_alt, 576p, 720p (best)
[cli][info] Opening stream: 720p (dash)
[cli][info] Starting player: "C:\Program Files\VideoLAN\VLC\vlc.exe"
[utils.named_pipe][info] Creating pipe streamlinkpipe-13756-1-9251
[utils.named_pipe][info] Creating pipe streamlinkpipe-13756-2-8365
Exception in thread Thread-DASHStreamWriter:
Traceback (most recent call last):
  File "C:\Users\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\segmented.py", line 209, in run
    self.write(segment, result, *data)
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 65, in write
    name = self._get_segment_name(segment)
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 29, in _get_segment_name
    return Path(urlparse(segment.url).path).resolve().name
  File "C:\Users\Python39\lib\pathlib.py", line 1215, in resolve
    s = self._flavour.resolve(self, strict=strict)
  File "C:\Users\Python39\lib\pathlib.py", line 215, in resolve
    s = self._ext_to_normal(_getfinalpathname(s))
OSError: [WinError 123] La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte: '\\eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjaXAiOiI4Ni4yNTIuMTgwLjI0NiIsImNtY2QiOiIiLCJleHAiOjE2OTc5OTYxNTIsImdpZCI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCIsImlhdCI6MTY5Nzk4MTc1MiwiaXNzIjoiZGVsaXZlcnkiLCJtYXhiIjowLCJzdGVtIjoiL291dC92MS83NTcwODhmN2ZkZjA0MWZmYTIxNDk0Zjc1ZDI3YzZjZSIsInN1YiI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCJ9.V49l3Q1kKepzqQaeEd6Zfr-RtlUgZYwZkXkIFCCxGZg'
Exception in thread Thread-DASHStreamWriter:
Traceback (most recent call last):
  File "C:\Users\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\segmented.py", line 209, in run
    self.write(segment, result, *data)
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 65, in write
    name = self._get_segment_name(segment)
  File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 29, in _get_segment_name
    return Path(urlparse(segment.url).path).resolve().name
  File "C:\Users\Python39\lib\pathlib.py", line 1215, in resolve
    s = self._flavour.resolve(self, strict=strict)
  File "C:\Users\Python39\lib\pathlib.py", line 215, in resolve
    s = self._ext_to_normal(_getfinalpathname(s))
OSError: [WinError 123] La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte: '\\eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjaXAiOiI4Ni4yNTIuMTgwLjI0NiIsImNtY2QiOiIiLCJleHAiOjE2OTc5OTYxNTIsImdpZCI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCIsImlhdCI6MTY5Nzk4MTc1MiwiaXNzIjoiZGVsaXZlcnkiLCJtYXhiIjowLCJzdGVtIjoiL291dC92MS83NTcwODhmN2ZkZjA0MWZmYTIxNDk0Zjc1ZDI3YzZjZSIsInN1YiI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCJ9.V49l3Q1kKepzqQaeEd6Zfr-RtlUgZYwZkXkIFCCxGZg'
[cli][error] Try 1/1: Could not open stream <DASHStream ['dash', 'https://live-lci-das.cdn-0.diff.tf1.fr/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjaXAiOiI4Ni4yNTIuMTgwLjI0NiIsImNtY2QiOiIiLCJleHAiOjE2OTc5OTYxNTIsImdpZCI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCIsImlhdCI6MTY5Nzk4MTc1MiwiaXNzIjoiZGVsaXZlcnkiLCJtYXhiIjowLCJzdGVtIjoiL291dC92MS83NTcwODhmN2ZkZjA0MWZmYTIxNDk0Zjc1ZDI3YzZjZSIsInN1YiI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCJ9.V49l3Q1kKepzqQaeEd6Zfr-RtlUgZYwZkXkIFCCxGZg/out/v1/757088f7fdf041ffa21494f75d27c6ce/index.mpd']> (No data returned from stream)
error: Could not open stream <DASHStream ['dash', 'https://live-lci-das.cdn-0.diff.tf1.fr/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjaXAiOiI4Ni4yNTIuMTgwLjI0NiIsImNtY2QiOiIiLCJleHAiOjE2OTc5OTYxNTIsImdpZCI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCIsImlhdCI6MTY5Nzk4MTc1MiwiaXNzIjoiZGVsaXZlcnkiLCJtYXhiIjowLCJzdGVtIjoiL291dC92MS83NTcwODhmN2ZkZjA0MWZmYTIxNDk0Zjc1ZDI3YzZjZSIsInN1YiI6ImU0YmUzMGUzLTdkNGYtNGJlMy1iMDk0LTgwZmY0NTY5YzNjNCJ9.V49l3Q1kKepzqQaeEd6Zfr-RtlUgZYwZkXkIFCCxGZg/out/v1/757088f7fdf041ffa21494f75d27c6ce/index.mpd']>, tried 1 times, exiting
[cli][info] Closing currently open stream...
```
While doing some research, I found a new command to open the TV channel on vlc or mpv. The problem is that I only have the video, not sound.

streamlink -p "C:\Program Files\VideoLAN\VLC\vlc.exe" https://www.tf1.fr/lci/direct 720p
streamlink -p mpv https://www.tf1.fr/lci/direct 720p

Any idea how to fix the problem ? My streamlink version : 5.0.1

Thanks and have a nice day.

**Comment by bastimeyer** [comment, marked as answer]
> A friend of mine installed streamlink on my computer over a year ago
> I hadn't been using it for a few months

> ```
> Exception in thread Thread-DASHStreamWriter:
> Traceback (most recent call last):
>   File "C:\Users\Python39\lib\threading.py", line 973, in _bootstrap_inner
>     self.run()
>   File "C:\Users\Python39\lib\site-packages\streamlink\stream\segmented.py", line 209, in run
>     self.write(segment, result, *data)
>   File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 65, in write
>     name = self._get_segment_name(segment)
>   File "C:\Users\Python39\lib\site-packages\streamlink\stream\dash.py", line 29, in _get_segment_name
>     return Path(urlparse(segment.url).path).resolve().name
>   File "C:\Users\Python39\lib\pathlib.py", line 1215, in resolve
>     s = self._flavour.resolve(self, strict=strict)
>   File "C:\Users\Python39\lib\pathlib.py", line 215, in resolve
>     s = self._ext_to_normal(_getfinalpathname(s))
> OSError: [WinError 123] La syntaxe du nom de fichier, de répertoire ou de volume est incorrecte: ...
> ```

Install the latest Streamlink version. According to the error trace, you must be on a version lower than `5.3.0`. The latest version is `6.2.1` with lots of changes in the DASH implementation since then. I can't reproduce this error on the latest version.

- https://streamlink.github.io/install.html

> The problem is that I only have the video, not sound.

Because you didn't set the path to FFmpeg on the command-line or in your Streamlink config file. FFmpeg is required for muxing separate video and audio streams, which is a requirement for DASH streams. Without FFmpeg (this can't be looked up automatically on Windows, because Windows...), only the video stream can be written to the output. Check the Streamlink documentation.

Since you don't appear to be using the Streamlink Windows installer, you will need to do this yourself. See the notes [here](https://streamlink.github.io/install.html#windows-binaries).

- https://streamlink.github.io/cli.html#ffmpeg-options
- https://streamlink.github.io/cli/config.html

**Comment by realpelou** [comment]
Hi,

Thank you for your answer.  I did the update and everything is OK now. I didn't want to do it until I had some advice.

Problem solved. :)
